weave/trace_server/clickhouse_trace_server_batched.py

```diff
@@ -766,118 +774,104 @@ def add_new_row_needed_to_insert(row_data: Any) -> str:
 
     def table_query(self, req: tsi.TableQueryReq) -> tsi.TableQueryRes:
         conds = []
-        parameters: Dict[str, Any] = {}
+        pb = ParamBuilder()
         if req.filter:
             if req.filter.row_digests:
-                conds.append("tr.digest IN {row_digests: Array(String)}")
-                parameters["row_digests"] = req.filter.row_digests
-        else:
-            conds.append("1 = 1")
+                conds.append(
+                    f"tr.digest IN {{{pb.add_param(req.filter.row_digests)}: Array(String)}}"
+                )
 
-        sort_clause = ""
-        pb = ParamBuilder()
+        sort_fields = []
         if req.sort_by:
-            sort_fields = []
             for i, sort in enumerate(req.sort_by):
                 # TODO: better splitting of escaped dots (.) in field names
                 extra_path = sort.field.split(".")
                 field = OrderField(
                     field=QueryBuilderDynamicField(
-                        field="val_dump", extra_path=extra_path
+                        field=VAL_DUMP_COLUMN_NAME, extra_path=extra_path
                     ),
                     direction="ASC" if sort.direction.lower() == "asc" else "DESC",
                 )
-                sort_fields.append(field.as_sql(pb, "tr"))
-            sort_clause = f"ORDER BY {', '.join(sort_fields)}"
-        sort_params = pb.get_params()
-        parameters.update(sort_params)
+                sort_fields.append(field)
         rows = self._table_query(
             req.project_id,
             req.digest,
-            conditions=conds,
+            pb,
+            sql_safe_conditions=conds,
+            sort_fields=sort_fields,
             limit=req.limit,
             offset=req.offset,
-            parameters=parameters,
-            sort_clause=sort_clause,
         )
         return tsi.TableQueryRes(rows=rows)
 
     def _table_query(
         self,
         project_id: str,
         digest: str,
-        conditions: Optional[list[str]] = None,
+        pb: ParamBuilder,
+        *,
+        sql_safe_conditions: Optional[list[str]] = None,
```

Comment:
Why did `conditions` change to `sql_safe_conditions`?  What does that mean?

Reply:
Added comment:
```
# using the `sql_safe_*` prefix is a way to signal to the caller
# that these strings should have been santized by the caller.
```